pkg/util/object-identifier.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2023 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/kubestellar/kubestellar/api/control/v1alpha1"
+)
+
+// ObjectIdentifier struct is used to add items to the workqueue.
+// The ObjectIdentifier contains all the necessary information to retrieve an object.
+type ObjectIdentifier struct {
+	GVK        schema.GroupVersionKind
+	Resource   string
+	ObjectName cache.ObjectName
+}
+
+func (identifier *ObjectIdentifier) GvkKey() string {
+	if identifier.GVK.Group == "" {
+		return fmt.Sprintf("%s/%s", identifier.GVK.Version, identifier.GVK.Kind)
+	}
+
+	return fmt.Sprintf("%s/%s/%s", identifier.GVK.Group, identifier.GVK.Version, identifier.GVK.Kind)
+}
+
+// RuntimeObjectToObjectIdentifier creates an ObjectIdentifier given an object
+// that implements runtime.Object. The gvkGvrMapper is used to retrieve the
+// resource name for the given runtime.Object.
+func RuntimeObjectToObjectIdentifier(rObj runtime.Object, gvkToGvrMapper GVKToGVRMapper) (ObjectIdentifier, error) {
+	ok := rObj.GetObjectKind()
+	gvk := ok.GroupVersionKind()
+
+	namespacedName, err := cache.ObjectToName(rObj)
+	if err != nil {
```

Comment:
If the input parameter were a `mrObject` instead of a `runtime.Object` then we could call https://github.com/kubernetes/client-go/blob/v0.28.2/tools/cache/store.go#L130 and not have to cope with the possibility of an error.